src/rendering.ts

```diff
@@ -511,7 +511,7 @@ export class StatusmapRenderer {
     } else if (this.panel.color.mode === 'spectrum') {
       return this.colorScale(d.value);
     } else if (this.panel.color.mode === 'discrete') {
-      if (this.panel.seriesFilterIndex != -1 || this.panel.seriesFilterIndex != null) {
+      if (this.panel.seriesFilterIndex != -1 && this.panel.seriesFilterIndex != null) {
```

Comment:
```suggestion
      if (this.panel.seriesFilterIndex != null && this.panel.seriesFilterIndex != -1) {
```

I think these two conditions should be swapped ;)